allennlp/modules/token_embedders/elmo_token_embedder_multilang.py

```diff
@@ -0,0 +1,171 @@
+    ----------
+    options_files : ``Dict[str, str]``, required.
+        A dictionary of language identifier to an ELMo JSON options file.
+    weight_files : ``Dict[str, str]``, required.
+        A dictionary of language identifier to an ELMo hdf5 weight file.
+    do_layer_norm : ``bool``, optional.
+        Should we apply layer normalization (passed to ``ScalarMix``)?
+    dropout : ``float``, optional.
+        The dropout value to be applied to the ELMo representations.
+    requires_grad : ``bool``, optional
+        If True, compute gradient of ELMo parameters for fine tuning.
+    projection_dim : ``int``, optional
+        If given, we will project the ELMo embedding down to this dimension.  We recommend that you
+        try using ELMo with a lot of dropout and no projection first, but we have found a few cases
+        where projection helps (particulary where there is very limited training data).
+    vocab_to_cache : ``List[str]``, optional, (default = 0.5).
+        A list of words to pre-compute and cache character convolutions
+        for. If you use this option, the ElmoTokenEmbedder expects that you pass word
+        indices of shape (batch_size, timesteps) to forward, instead
+        of character indices. If you use this option and pass a word which
+        wasn't pre-cached, this will break.
+    scalar_mix_parameters : ``List[int]``, optional, (default=None).
+        If not ``None``, use these scalar mix parameters to weight the representations
+        produced by different layers. These mixing weights are not updated during
+        training.
+    aligning_files : ``Dict[str, str]``, optional, (default={}).
+        A dictionary of language identifier to a pth file with an alignment matrix.
+    """
+
+    def __init__(self,
+                 options_files: Dict[str, str],
+                 weight_files: Dict[str, str],
+                 do_layer_norm: bool = False,
+                 dropout: float = 0.5,
+                 requires_grad: bool = False,
+                 projection_dim: int = None,
+                 vocab_to_cache: List[str] = None,
+                 scalar_mix_parameters: List[float] = None,
+                 aligning_files: Dict[str, str] = {}) -> None:
+        super(ElmoTokenEmbedderMultiLang, self).__init__()
+
+        assert options_files.keys() == weight_files.keys()
+        output_dim = -1
+        for lang in weight_files.keys():
+            name = 'elmo_%s' % lang
+            elmo = Elmo(
+                    options_files[lang],
+                    weight_files[lang],
+                    1,
+                    do_layer_norm=do_layer_norm,
+                    dropout=dropout,
+                    requires_grad=requires_grad,
+                    vocab_to_cache=vocab_to_cache,
+                    scalar_mix_parameters=scalar_mix_parameters)
+            self.add_module(name, elmo)
+
+            output_dim_tmp = elmo.get_output_dim()
+            if output_dim != -1:
```

Comment:
Safer and more intuitive to have this be initialized to `None` instead of `-1` (note that pylint will want you to do `is not None` instead of `!= None`).